system/lib/emmalloc.c

```diff
@@ -524,23 +524,22 @@ static bool claim_more_memory(size_t numBytes)
   return true;
 }
 
-// Initialize malloc during static initialization with highest constructor priority,
-// so that it initializes before any other static initializers in compilation units.
-static void EMSCRIPTEN_KEEPALIVE __attribute__((constructor(0))) initialize_malloc_heap()
+// The current sequence of system-level static init functions in emscripten is:
+// 47: emmalloc_init
+// 48: __emscripten_pthread_data_constructor
+// 49: emscripten_tls_init
+// 50: asan init
+// (Please keep the above list updated alongside each one of these fuctions).
+static void __attribute__((constructor(47))) emmalloc_init()
```

Comment:
The reason to name this function differently was to separate that it is not part of the public `emmalloc_*()` API. Having `emmalloc_init()` makes it read like it would be at the same API level as `emmalloc_malloc()` et al.

It is marked static - but still the rename takes it closer to emmalloc_* public API. Maybe `_emmalloc_init()` or `static_emmalloc_init()` or similar so readers of the file won't group it together with the public API?

Reply:
Renamed